### PR TITLE
[Keyboard] Added Frooastboard

### PR DESCRIPTION
**Pull request by Frooastside**
## Description

I added my own Keyboard to QMK.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Added Frooastboard

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/frooastboard/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/frooastboard/keymaps/via/config.h:19]: could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

Also, in the via json you can specify which modes are enabled or not.

File: keyboards/frooastboard/rules.mk [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+#     If no clock division is performed on the input clock inside the AVR (via the
+#     CPU clock adjust registers or the clock division fuses), this will be equal to F_CPU.
+F_USB = $(F_CPU)
+
+# Interrupt driven control endpoint task(+60)
+OPT_DEFS += -DINTERRUPT_CONTROL_ENDPOINT
+
+
+# Boot Section Size in *bytes*
+OPT_DEFS += -DBOOTLOADER_SIZE=4096
```
Comment (drashna) [inline keyboards/frooastboard/rules.mk:43]:
```suggestion
```

File: keyboards/frooastboard/rules.mk [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+MOUSEKEY_ENABLE ?= no      # Mouse keys(+4700)
+EXTRAKEY_ENABLE ?= no      # Audio control and System control(+450)
+CONSOLE_ENABLE ?= no       # Console for debug(+400)
+COMMAND_ENABLE ?= no       # Commands for debug and configuration
+SLEEP_LED_ENABLE ?= no     # Breathing sleep LED during USB suspend
+NKRO_ENABLE ?= yes         # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE ?= no     # Enable keyboard backlight functionality
+AUDIO_ENABLE ?= no         # Audio output
+BLUETOOTH_ENABLE = no      # Enable Bluetooth
+RGBLIGHT_ENABLE ?= yes     # Enable keyboard RGB underglow
```
Comment (drashna) [inline keyboards/frooastboard/rules.mk:59]:
```suggestion
BOOTMAGIC_ENABLE = lite   # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no      # Mouse keys
EXTRAKEY_ENABLE = no      # Audio control and System control
CONSOLE_ENABLE = no       # Console for debug
COMMAND_ENABLE = no       # Commands for debug and configuration
SLEEP_LED_ENABLE = no     # Breathing sleep LED during USB suspend
NKRO_ENABLE = yes         # USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no     # Enable keyboard backlight functionality
AUDIO_ENABLE = no         # Audio output
BLUETOOTH_ENABLE = no     # Enable Bluetooth
RGBLIGHT_ENABLE = yes     # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/frooastboard/rules.mk [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u2
+
+BOOTLOADER = atmel-dfu
+
+
+
```
Comment (drashna) [inline keyboards/frooastboard/rules.mk:7]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu

```

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**